### DISCUSSION TITLE
How let user make own fields in entity

### DISCUSSION THREAD

**Discussion by cube4d** [symfony/symfony#46541] [Q&A] [answered]
Hi,

i am building a webapp where a user can save a date in an entity.
As an example i have a truck entity where the user can put all his trucks (transport company)
and then have some fields like when is this trucks MOT ? (i think thats the englisch version)
or when is the next workshop appointment. 

Then the user get a warning message one day or one month before.  

Now i want to let the user create his own field in this entity and let him select his own time before the warning, 
like 1 week before or 2 days before...

But this/my entity is for all users and i dont like to let the users create 1000... new fields, i dont like that idea
and i think there will be too much new fields in the future.
Do you guys have a better idea or do you know how other apps do that ?
My last idea was to create an array field "custom" and there save the date with the selected warning time ?
But that seems to be really complicated when i will show a table or so or do all the CRUD stuff.

Thanks

**Comment by lyrixx** [comment, marked as answer]
If there is only one warning, you can create a generic "warningIn" fields.

If you allow user to create many warning you have two options:

1. A many to [many relation](https://www.doctrine-project.org/projects/doctrine-orm/en/2.9/reference/association-mapping.html). Like that, user can add as many warning as they want. This is the simplest solution, but with more code
2. Add a `warningsIn` fields as JSON column. But you must use a DB that is powerful with JSON (like PostgreSQL) and you need some skills with SQL + JSON. I would go with this one, since it requires less table, and less code.

**Reply by cube4d** [reply]
Ok thank you very much i will try my best :-)
